src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -2085,6 +2085,11 @@ public override BoundStatement BindStatement(StatementSyntax node, DiagnosticBag
                 return (BoundStatement)boundNode;
             }
 
+            internal override BoundBlock BindEmbeddedBlock(BlockSyntax node, DiagnosticBag diagnostics)
+            {
+                return (BoundBlock)TryGetBoundNodeFromMap(node) ?? base.BindEmbeddedBlock(node, diagnostics);
```

Comment:
Will the `base` call properly populate the cache?

Reply:
Cache population is done after binding, not here.

---
In reply to: [227889774](https://github.com/dotnet/roslyn/pull/30721#discussion_r227889774) [](ancestors = 227889774)